<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#OVERVIEW-" data-toc-modified-id="OVERVIEW--1"><span class="toc-item-num">1&nbsp;&nbsp;</span>OVERVIEW <br></a></span><ul class="toc-item"><li><span><a href="#Goal" data-toc-modified-id="Goal-1.1"><span class="toc-item-num">1.1&nbsp;&nbsp;</span>Goal</a></span></li><li><span><a href="#Steps" data-toc-modified-id="Steps-1.2"><span class="toc-item-num">1.2&nbsp;&nbsp;</span>Steps</a></span></li></ul></li><li><span><a href="#API-" data-toc-modified-id="API--2"><span class="toc-item-num">2&nbsp;&nbsp;</span>API <br></a></span><ul class="toc-item"><li><span><a href="#Preparing-movie-title-list" data-toc-modified-id="Preparing-movie-title-list-2.1"><span class="toc-item-num">2.1&nbsp;&nbsp;</span>Preparing movie title list</a></span></li><li><span><a href="#Extracting-full-info-about-each-movie-from-the-generated-list" data-toc-modified-id="Extracting-full-info-about-each-movie-from-the-generated-list-2.2"><span class="toc-item-num">2.2&nbsp;&nbsp;</span>Extracting full info about each movie from the generated list</a></span></li><li><span><a href="#Cleaning-Dataset" data-toc-modified-id="Cleaning-Dataset-2.3"><span class="toc-item-num">2.3&nbsp;&nbsp;</span>Cleaning Dataset</a></span></li><li><span><a href="#Saving-results-to-CSV" data-toc-modified-id="Saving-results-to-CSV-2.4"><span class="toc-item-num">2.4&nbsp;&nbsp;</span>Saving results to CSV</a></span></li><li><span><a href="#Saving-poster-images" data-toc-modified-id="Saving-poster-images-2.5"><span class="toc-item-num">2.5&nbsp;&nbsp;</span>Saving poster images</a></span></li></ul></li><li><span><a href="#WEB-SCRAPING-" data-toc-modified-id="WEB-SCRAPING--3"><span class="toc-item-num">3&nbsp;&nbsp;</span>WEB SCRAPING <br></a></span><ul class="toc-item"><li><span><a href="#Preparing-the-list-of-IMDB-IDs" data-toc-modified-id="Preparing-the-list-of-IMDB-IDs-3.1"><span class="toc-item-num">3.1&nbsp;&nbsp;</span>Preparing the list of IMDB IDs</a></span></li><li><span><a href="#Extracting-Director,-Title,-Rating,-Year,-Videos-Link-from-IMDB" data-toc-modified-id="Extracting-Director,-Title,-Rating,-Year,-Videos-Link-from-IMDB-3.2"><span class="toc-item-num">3.2&nbsp;&nbsp;</span>Extracting Director, Title, Rating, Year, Videos Link from IMDB</a></span></li><li><span><a href="#Saving-result-to-CSV" data-toc-modified-id="Saving-result-to-CSV-3.3"><span class="toc-item-num">3.3&nbsp;&nbsp;</span>Saving result to CSV</a></span></li></ul></li><li><span><a href="#DATA-CHECK-" data-toc-modified-id="DATA-CHECK--4"><span class="toc-item-num">4&nbsp;&nbsp;</span>DATA CHECK <br></a></span><ul class="toc-item"><li><span><a href="#Check-why-in-two-columns-have-different-year-(data-from-API)" data-toc-modified-id="Check-why-in-two-columns-have-different-year-(data-from-API)-4.1"><span class="toc-item-num">4.1&nbsp;&nbsp;</span>Check why in two columns have different year (data from API)</a></span></li><li><span><a href="#API-data-vs-web-scraping-IMDB" data-toc-modified-id="API-data-vs-web-scraping-IMDB-4.2"><span class="toc-item-num">4.2&nbsp;&nbsp;</span>API data vs web scraping IMDB</a></span></li><li><span><a href="#Creating-one-dataframe-with-the-most-important-info-from-API-and-IMDB,-saving-to-CSV" data-toc-modified-id="Creating-one-dataframe-with-the-most-important-info-from-API-and-IMDB,-saving-to-CSV-4.3"><span class="toc-item-num">4.3&nbsp;&nbsp;</span>Creating one dataframe with the most important info from API and IMDB, saving to CSV</a></span></li></ul></li><li><span><a href="#CORRELATION-BETWEEN-RATING-AND-VALENTINE'S-DAY-" data-toc-modified-id="CORRELATION-BETWEEN-RATING-AND-VALENTINE'S-DAY--5"><span class="toc-item-num">5&nbsp;&nbsp;</span>CORRELATION BETWEEN RATING AND VALENTINE'S DAY <br></a></span><ul class="toc-item"><li><span><a href="#Getting-the-number-of-days-to-Valentine's-day-per-movie" data-toc-modified-id="Getting-the-number-of-days-to-Valentine's-day-per-movie-5.1"><span class="toc-item-num">5.1&nbsp;&nbsp;</span>Getting the number of days to Valentine's day per movie</a></span></li><li><span><a href="#Binning-the-data-based-on-the-days-to-Valentine's-day" data-toc-modified-id="Binning-the-data-based-on-the-days-to-Valentine's-day-5.2"><span class="toc-item-num">5.2&nbsp;&nbsp;</span>Binning the data based on the days to Valentine's day</a></span></li><li><span><a href="#Checking-the-correlation-between-the-rating-and-the-days-to-Valentine's-day" data-toc-modified-id="Checking-the-correlation-between-the-rating-and-the-days-to-Valentine's-day-5.3"><span class="toc-item-num">5.3&nbsp;&nbsp;</span>Checking the correlation between the rating and the days to Valentine's day</a></span></li><li><span><a href="#Checking-the-%-of-the-movies-that-have-been-release-very-close---close-to-Valentine's-day" data-toc-modified-id="Checking-the-%-of-the-movies-that-have-been-release-very-close---close-to-Valentine's-day-5.4"><span class="toc-item-num">5.4&nbsp;&nbsp;</span>Checking the % of the movies that have been release very close - close to Valentine's day</a></span></li></ul></li><li><span><a href="#Results" data-toc-modified-id="Results-6"><span class="toc-item-num">6&nbsp;&nbsp;</span>Results</a></span></li></ul></div>

## OVERVIEW <br>

### Goal
- Collect the information about the movies that have 'love' in the title for the Valentine's day campaign (Title, Director, Rating, Release Year, Link to Trailers and Videos, download Posters)
- Check if the information from the API is coinciding with the information in IMDB
- Check the correlation between the days to Valentines day and rating

### Steps

- Use OMDB API: http://www.omdbapi.com/#usage
- Do the API request with the search, movies with word 'love', extract information from 4 pages
- Generate the list of titles of the movies
- Do another API request title by title to get all information about the movies
- Put info from both requests to one dataframe
- Scrape IMDB based on IMDB ID to get the main information
- Check if data is matching in the dataframes from API and IMDB in these columns: Title, Year, Director, IMDB Rating
- Prepare one dataframe with information from API and IMDB and export
- Analyze if there is any correlation between the Release Day days to Valentine's day and the rating value

## API <br>

For the API portion of the project will need to make calls to your chosen API, successfully obtain a response, request data, convert it into a Pandas data frame, and export it as a CSV file.

In [1]:
from pandas.io.json import json_normalize
import pandas as pd
import json
import requests
from bs4 import BeautifulSoup
import urllib.request
from urllib.request import urlopen
import re
import numpy as np
import time

In [2]:
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

In [3]:
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)
pd.set_option('display.width', None)
pd.set_option('display.max_colwidth', -1)

### Preparing movie title list

- Extracting titles that include word 'love', loading 4 pages
- Converting names into the correct format to later pass them to the URL

In [4]:
#in the range indicate how many pages want to check (API in general gives MAX 10 movies per search)
#adding all the movie names to one list
#per each loop generating json for one page and extracting titles
#changing titles to the correct format to later pass to the URL

def formatted_titles_api(url_pat):
    """
    Function takes the URL pattern and returns the formatted list of movie titles, ready to be used with another request.
    Can indicate a specific number of pages to return
    
    Input:
        url_pat: the pattern of the URL
    Output:
        titles_formatted: the list of the movie titles with a specific format
    """
    final_list = []
    for page in range(1,5):
        page = str(page)
        url = url_pat + page
        response = requests.get(url)
        results = response.json()
        title_box = results.get('Search')
        title_list = [title.get('Title') for title in title_box]
        final_list += title_list
        titles_formatted = [string.replace(" ", "+") for string in final_list]
    return titles_formatted
        

In [5]:
#calling the function

url_pat = 'http://www.omdbapi.com/?i=tt3896198&apikey=aab3947e&s=love&type=movie&page='
titles_formatted = formatted_titles_api(url_pat)
titles_formatted

['Crazy,+Stupid,+Love.',
 'Dr.+Strangelove+or:+How+I+Learned+to+Stop+Worrying+and+Love+the+Bomb',
 'Love,+Actually',
 'Shakespeare+in+Love',
 'P.S.+I+Love+You',
 'I+Love+You,+Man',
 'Love+&+Other+Drugs',
 'Punch-Drunk+Love',
 'In+the+Mood+for+Love',
 'Love,+Rosie',
 'From+Russia+with+Love',
 'From+Paris+with+Love',
 'Love,+Simon',
 'I+Love+You+Phillip+Morris',
 'Eat+Pray+Love',
 'To+Rome+with+Love',
 'Stuck+in+Love.',
 'A+Lot+Like+Love',
 'Love+Me+If+You+Dare',
 'Wristcutters:+A+Love+Story',
 'The+Love+Guru',
 'Show+Me+Love',
 'New+York,+I+Love+You',
 'American+Pie+Presents:+The+Book+of+Love',
 'Love',
 'Capitalism:+A+Love+Story',
 'Endless+Love',
 'Down+with+Love',
 'Sea+of+Love',
 'Everyone+Says+I+Love+You',
 'Love+and+Death',
 'Love+&+Mercy',
 'I+Love+You,+Beth+Cooper',
 'The+One+I+Love',
 "Can't+Buy+Me+Love",
 'For+Love+of+the+Game',
 'All+the+Boys+Love+Mandy+Lane',
 'Love+Happens',
 'Must+Love+Dogs',
 'Love+Story']

### Extracting full info about each movie from the generated list

In [6]:
def movies_info_api(url_pattern):
    """
    Function loops through the urls and returns one dataframe with the information of all collected movies
    
    Input:
        url_pattern: the pattern of the URL
    Output:
        love_movies: the dataframe with the information of all collected movies
    """
    appended_dfs=[]
    for title in titles_formatted:
        url = url_pattern + title
        response = requests.get(url)
        results = response.json()
        flattened_data = json_normalize(results)
        #appending dataframes to the list to later concatenate them to one dataframe
        appended_dfs.append(flattened_data)
        love_movies = pd.concat(appended_dfs, ignore_index = True) #ignoring index, to have 0,1,2...n, if not, get 0,0,0,...

    return love_movies

In [7]:
#calling the function

url_pattern = 'http://www.omdbapi.com/?i=tt3896198&apikey=aab3947e&t='
love_movies = movies_info_api(url_pattern)
love_movies.head()

Title  Year  \
0  Crazy, Stupid, Love.                                                  2011   
1  Dr. Strangelove or: How I Learned to Stop Worrying and Love the Bomb  1964   
2  Love, Actually                                                        2003   
3  Shakespeare in Love                                                   1998   
4  P.S. I Love You                                                       2007   

   Rated     Released  Runtime                            Genre  \
0  PG-13  29 Jul 2011  118 min  Comedy, Drama, Romance            
1  PG     29 Jan 1964  95 min   Comedy                            
2  R      14 Nov 2003  135 min  Comedy, Drama, Romance            
3  R      08 Jan 1999  123 min  Comedy, Drama, History, Romance   
4  PG-13  21 Dec 2007  126 min  Comedy, Drama, Romance            

                    Director  \
0  Glenn Ficarra, John Requa   
1  Stanley Kubrick             
2  Richard Curtis              
3  John Madden                 
4  Richard LaGravenese         

                                                                                                                                   Writer  \
0  Dan Fogelman                                                                                                                             
1  Stanley Kubrick (screenplay), Terry Southern (screenplay), Peter George (screenplay), Peter George (based on the book: "Red Alert" by)   
2  Richard Curtis                                                                                                                           
3  Marc Norman, Tom Stoppard                                                                                                                
4  Richard LaGravenese (screenplay), Steven Rogers (screenplay), Cecelia Ahern (novel)                                                      

                                                         Actors  \
0  Steve Carell, Ryan Gosling, Julianne Moore, Emma Stone         
1  Peter Sellers, George C. Scott, Sterling Hayden, Keenan Wynn   
2  Bill Nighy, Gregor Fisher, Rory MacGregor, Colin Firth         
3  Geoffrey Rush, Tom Wilkinson, Steven O'Donnell, Tim McMullan   
4  Hilary Swank, Gerard Butler, Lisa Kudrow, Gina Gershon         

                                                                                                                                                                                                       Plot  \
0  A middle-aged husband's life changes dramatically when his wife asks him for a divorce. He seeks to rediscover his manhood with the help of a newfound friend, Jacob, learning to pick up girls at bars.   
1  An insane general triggers a path to nuclear holocaust that a War Room full of politicians and generals frantically tries to stop.                                                                         
2  Follows the lives of eight very different couples in dealing with their love lives in various loosely interrelated tales all set during a frantic month before Christmas in London, England.               
3  The world's greatest ever playwright, William Shakespeare, is young, out of ideas and short of cash, but meets his ideal woman and is inspired to write one of his most famous plays.                      
4  A young widow discovers that her late husband has left her 10 messages intended to help ease her pain and start a new life.                                                                                

                      Language          Country  \
0  English                      USA               
1  English, Russian             UK, USA           
2  English, Portuguese, French  UK, USA, France   
3  English                      USA, UK           
4  English                      USA               

                                                             Awards  \
0  Nominated for 1 Golden Globe. Another 3 wins & 22 nominations.     
1  Nominated for 4 Oscars. Another 13 win

### Cleaning Dataset
- Changing column names
- Checking for NaNs
- Deleting unnecesary columns

In [8]:
# changing column names to lowercase and to snake
# changing column namr 'reñeased' to 'released_spain' to be more specific

love_movies.columns

love_movies.rename(columns=lambda x: x.strip().lower(), inplace=True)
colnames = ['title', 'year', 'rated', 'released_spain', 'runtime', 'genre', 'director',
       'writer', 'actors', 'plot', 'language', 'country', 'awards', 'poster',
       'ratings', 'metascore', 'imdb_rating', 'imdb_votes', 'imdb_id', 'type',
       'dvd', 'boxoffice', 'production', 'website', 'response']

love_movies.columns = colnames

Index(['Title', 'Year', 'Rated', 'Released', 'Runtime', 'Genre', 'Director',
       'Writer', 'Actors', 'Plot', 'Language', 'Country', 'Awards', 'Poster',
       'Ratings', 'Metascore', 'imdbRating', 'imdbVotes', 'imdbID', 'Type',
       'DVD', 'BoxOffice', 'Production', 'Website', 'Response'],
      dtype='object')

In [9]:
#changing N/A to NaN

love_movies = love_movies.replace('N/A', np.nan)

In [10]:
#total number of null cells per column

null_cols = love_movies.isnull().sum() 
null_cols[null_cols > 0]

awards        3 
metascore     2 
boxoffice     18
production    2 
website       40
dtype: int64

In [11]:
#deleting column website, all rows empty
#if decided that need boxoffice, could scrape it from wikipedia, now don't do anything

love_movies = love_movies.drop(['website'],axis=1)

In [12]:
love_movies.head()

title  year  \
0  Crazy, Stupid, Love.                                                  2011   
1  Dr. Strangelove or: How I Learned to Stop Worrying and Love the Bomb  1964   
2  Love, Actually                                                        2003   
3  Shakespeare in Love                                                   1998   
4  P.S. I Love You                                                       2007   

   rated released_spain  runtime                            genre  \
0  PG-13  29 Jul 2011    118 min  Comedy, Drama, Romance            
1  PG     29 Jan 1964    95 min   Comedy                            
2  R      14 Nov 2003    135 min  Comedy, Drama, Romance            
3  R      08 Jan 1999    123 min  Comedy, Drama, History, Romance   
4  PG-13  21 Dec 2007    126 min  Comedy, Drama, Romance            

                    director  \
0  Glenn Ficarra, John Requa   
1  Stanley Kubrick             
2  Richard Curtis              
3  John Madden                 
4  Richard LaGravenese         

                                                                                                                                   writer  \
0  Dan Fogelman                                                                                                                             
1  Stanley Kubrick (screenplay), Terry Southern (screenplay), Peter George (screenplay), Peter George (based on the book: "Red Alert" by)   
2  Richard Curtis                                                                                                                           
3  Marc Norman, Tom Stoppard                                                                                                                
4  Richard LaGravenese (screenplay), Steven Rogers (screenplay), Cecelia Ahern (novel)                                                      

                                                         actors  \
0  Steve Carell, Ryan Gosling, Julianne Moore, Emma Stone         
1  Peter Sellers, George C. Scott, Sterling Hayden, Keenan Wynn   
2  Bill Nighy, Gregor Fisher, Rory MacGregor, Colin Firth         
3  Geoffrey Rush, Tom Wilkinson, Steven O'Donnell, Tim McMullan   
4  Hilary Swank, Gerard Butler, Lisa Kudrow, Gina Gershon         

                                                                                                                                                                                                       plot  \
0  A middle-aged husband's life changes dramatically when his wife asks him for a divorce. He seeks to rediscover his manhood with the help of a newfound friend, Jacob, learning to pick up girls at bars.   
1  An insane general triggers a path to nuclear holocaust that a War Room full of politicians and generals frantically tries to stop.                                                                         
2  Follows the lives of eight very different couples in dealing with their love lives in various loosely interrelated tales all set during a frantic month before Christmas in London, England.               
3  The world's greatest ever playwright, William Shakespeare, is young, out of ideas and short of cash, but meets his ideal woman and is inspired to write one of his most famous plays.                      
4  A young widow discovers that her late husband has left her 10 messages intended to help ease her pain and start a new life.                                                                                

                      language          country  \
0  English                      USA               
1  English, Russian             UK, USA           
2  English, Portuguese, French  UK, USA, France   
3  English                      USA, UK           
4  English                      USA               

                                                             awards  \
0  Nominated for 1 Golden Globe. Another 3 wins & 22 nominations.     
1  Nominated for 4 Oscars. An

### Saving results to CSV

In [13]:
love_movies.to_csv("/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/love_movies_api.csv", index=False)

### Saving poster images

In [14]:
movie_url = love_movies['poster'].to_list()
image_name = love_movies['title'].to_list()

for url, name in zip(movie_url, image_name):
    urllib.request.urlretrieve(url, '/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/{}.jpg'.format(name))


('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/Crazy, Stupid, Love..jpg',
 <http.client.HTTPMessage at 0x1f0c1f0de88>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/Dr. Strangelove or: How I Learned to Stop Worrying and Love the Bomb.jpg',
 <http.client.HTTPMessage at 0x1f0c1ee3108>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/Love, Actually.jpg',
 <http.client.HTTPMessage at 0x1f0bebbaf88>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/Shakespeare in Love.jpg',
 <http.client.HTTPMessage at 0x1f0c1f29788>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/P.S. I Love You.jpg',
 <http.client.HTTPMessage at 0x1f0c1f147c8>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/I Love You, Man.jpg',
 <http.client.HTTPMessage at 0x1f0c1f3b9c8>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/Love.jpg',
 <http.client.HTTPMessage at 0x1f0c1f0d9c8>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/Punch-Drunk Love.jpg',
 <http.client.HTTPMessage at 0x1f0c1e93e88>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/In the Mood for Love.jpg',
 <http.client.HTTPMessage at 0x1f0c1f29488>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/Love, Rosie.jpg',
 <http.client.HTTPMessage at 0x1f0c1f29788>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/From Russia with Love.jpg',
 <http.client.HTTPMessage at 0x1f0c1f2c4c8>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/From Paris with Love.jpg',
 <http.client.HTTPMessage at 0x1f0c1f2ccc8>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/Love, Simon.jpg',
 <http.client.HTTPMessage at 0x1f0c1e93e08>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/I Love You Phillip Morris.jpg',
 <http.client.HTTPMessage at 0x1f0c1f29a48>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/Eat Pray Love.jpg',
 <http.client.HTTPMessage at 0x1f0c1f29b08>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/To Rome with Love.jpg',
 <http.client.HTTPMessage at 0x1f0c1f2c088>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/Stuck in Love..jpg',
 <http.client.HTTPMessage at 0x1f0c1f2c388>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/A Lot Like Love.jpg',
 <http.client.HTTPMessage at 0x1f0c1f2ce48>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/Love Me If You Dare.jpg',
 <http.client.HTTPMessage at 0x1f0c1f0dd48>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/Wristcutters: A Love Story.jpg',
 <http.client.HTTPMessage at 0x1f0c1f29108>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/The Love Guru.jpg',
 <http.client.HTTPMessage at 0x1f0c1f29888>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/Show Me Love.jpg',
 <http.client.HTTPMessage at 0x1f0c1f3b9c8>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/New York, I Love You.jpg',
 <http.client.HTTPMessage at 0x1f0c1f2c788>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/American Pie Presents: The Book of Love.jpg',
 <http.client.HTTPMessage at 0x1f0c1f2cd88>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/Love.jpg',
 <http.client.HTTPMessage at 0x1f0c1f0dbc8>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/Capitalism: A Love Story.jpg',
 <http.client.HTTPMessage at 0x1f0c1f29208>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/Endless Love.jpg',
 <http.client.HTTPMessage at 0x1f0c1f29888>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/Down with Love.jpg',
 <http.client.HTTPMessage at 0x1f0c1f3b5c8>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/Sea of Love.jpg',
 <http.client.HTTPMessage at 0x1f0c1f2cfc8>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/Everyone Says I Love You.jpg',
 <http.client.HTTPMessage at 0x1f0c1f14148>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/Love and Death.jpg',
 <http.client.HTTPMessage at 0x1f0c1f29708>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/Love.jpg',
 <http.client.HTTPMessage at 0x1f0c1f29848>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/I Love You, Beth Cooper.jpg',
 <http.client.HTTPMessage at 0x1f0c1eb8348>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/The One I Love.jpg',
 <http.client.HTTPMessage at 0x1f0c1f0d908>)

("/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/Can't Buy Me Love.jpg",
 <http.client.HTTPMessage at 0x1f0c1f2cfc8>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/For Love of the Game.jpg',
 <http.client.HTTPMessage at 0x1f0c1f2cbc8>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/All the Boys Love Mandy Lane.jpg',
 <http.client.HTTPMessage at 0x1f0c1f29b48>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/Love Happens.jpg',
 <http.client.HTTPMessage at 0x1f0c1f29948>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/Must Love Dogs.jpg',
 <http.client.HTTPMessage at 0x1f0c1f29708>)

('/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/image_folder/Love Story.jpg',
 <http.client.HTTPMessage at 0x1f0c1f2c308>)

## WEB SCRAPING <br>

For the web scraping portion of the project, you will need to scrape the HTML from your chosen page, parse the HTML to extract the necessary information, and either save the results to a text (txt) file if it is text or into a CSV file if it is tabular data.

### Preparing the list of IMDB IDs

In [15]:
#taking movie IDs from API dataframe, to use them for scraping IMDB

id_list = list(love_movies['imdb_id'])

### Extracting Director, Title, Rating, Year, Videos Link from IMDB

In [16]:
#creating function to scrape data from IMDB

def scraped_info_to_dataframe(url_pattern):
    """
    Function loops through the urls and returns one dataframe with the information of all the movies that were checked via API
    There is a sleep time of 2s for each request.
    
    Input:
        url_pattern: the pattern of the URL
    Output:
        love_movies_imdb: the dataframe with the information of the selected movies 
                          (director, title, rating, year, links to trailers)
    """
    info_final = []

    for m_id in id_list:
        url = url_pattern + m_id
        time.sleep(2) #sleep time, could be changed to any
        html = urlopen(url)
        soup = BeautifulSoup(html, 'lxml')

        #extracting director
        summary = soup.find_all('div', {'class':'credit_summary_item'})
        director = summary[0].get_text().strip().split('\n')[1]

        #extracting title
        title_box = soup.find_all('h1', {'class':''})
        for t in title_box:
            title = t.text.split('\xa0')[0]
        #if title is in Spanish, means that original title class exists, then take the title from there, to have it in English    
        original_title_box = soup.find_all('div', {'class':'originalTitle'})
        if original_title_box:
            title_long = [t.text.split('\xa0')[0] for t in original_title_box]
            pattern = re.compile("(.*?)\s*\((.*?)\)") #only taking the string part before the ()
            for string in title_long:
                title = pattern.match(string).group(1)

        #extracting rating
        rating_box = soup.find_all('div', {'class':'ratingValue'})
        rates = [rate.text.strip() for rate in rating_box]
        pattern = '\d+\D\d' #extracting format only x.x
        for ratio in rates:
            rating = re.findall(pattern,ratio)[0]

        #extracting year
        year_box = soup.find_all('h1', {'class':''})
        year_long = [y.text.strip().split('\xa0')[1] for y in year_box]
        pattern = re.compile(".*?\((.*?)\)") #extrcting year from ()
        for string in year_long:
            year = pattern.match(string).group(1)
        
        #adding links to videos per movie
        videos = 'https://www.imdb.com/title/x/videogallery/?ref_=tt_ql_pv_2'
        for m_id in id_list:
            videos = videos.replace('x', m_id)
        
        #adding all info to one nested list
        info_final += [[title, year, rating, director, videos]]

    #creating dataframe with the scraped info
    colnames = ['title', 'year', 'imdb_rating','director', 'video_link']
    love_movies_imdb = pd.DataFrame(info_final)
    love_movies_imdb.columns = colnames
    
    return love_movies_imdb


In [17]:
#calling the function

url_pattern = 'https://www.imdb.com/title/'

love_movies_imdb = scraped_info_to_dataframe(url_pattern)
love_movies_imdb.head()

title  year  \
0  Crazy, Stupid, Love.                                                  2011   
1  Dr. Strangelove or: How I Learned to Stop Worrying and Love the Bomb  1964   
2  Love Actually                                                         2003   
3  Shakespeare in Love                                                   1998   
4  P.S. I Love You                                                       2007   

  imdb_rating                   director  \
0  7.4         Glenn Ficarra, John Requa   
1  8.4         Stanley Kubrick             
2  7.6         Richard Curtis              
3  7.1         John Madden                 
4  7.0         Richard LaGravenese         

                                                           video_link  
0  https://www.imdb.com/title/tt1570728/videogallery/?ref_=tt_ql_pv_2  
1  https://www.imdb.com/title/tt1570728/videogallery/?ref_=tt_ql_pv_2  
2  https://www.imdb.com/title/tt1570728/videogallery/?ref_=tt_ql_pv_2  
3  https://www.imdb.com/title/tt1570728/videogallery/?ref_=tt_ql_pv_2  
4  https://www.imdb.com/title/tt1570728/videogallery/?ref_=tt_ql_pv_2

### Saving result to CSV

In [18]:
love_movies_imdb.to_csv("/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/love_movies_imdb.csv", index=False)

## DATA CHECK <br>

Check if data scraped from IMDB matches with the data from API

### Check why in two columns have different year (data from API)

- Checking why and in which movies have different year

In [19]:
#changing column type to datetime

love_movies['released_spain'] = pd.to_datetime(love_movies['released_spain']) 

In [20]:
#extracting year from release to another column

love_movies['year_from_released'] = love_movies['released_spain'].dt.year

In [21]:
#changing type to int to be able to compare

love_movies['year'] = love_movies['year'].astype('int')

In [22]:
#checking where year is not equal to the year from release date

df = love_movies[['year', 'year_from_released', 'title', 'imdb_id']]
df.loc[df['year'] != df['year_from_released']]


year  year_from_released                         title    imdb_id
3   1998  1999                Shakespeare in Love           tt0138097
8   2000  2001                In the Mood for Love          tt0118694
10  1963  1964                From Russia with Love         tt0057076
13  2009  2011                I Love You Phillip Morris     tt1045772
16  2012  2013                Stuck in Love.                tt2205697
18  2003  2004                Love Me If You Dare           tt0364517
19  2006  2007                Wristcutters: A Love Story    tt0477139
22  2008  2009                New York, I Love You          tt0808399
29  1996  1997                Everyone Says I Love You      tt0116242
36  2006  2013                All the Boys Love Mandy Lane  tt0490076

After checking somae cases directly in IMDB understood that Year from Released is the release year in Spain

### API data vs web scraping IMDB

In [23]:
#changing 'year' to int and 'imdb_rating' to float in df from imdb to be able to do the comparison

love_movies_imdb['year'] = love_movies_imdb['year'].astype('int')
love_movies_imdb['imdb_rating'] = love_movies_imdb['imdb_rating'].astype('float')
love_movies['imdb_rating'] = love_movies['imdb_rating'].astype('float')

In [24]:
#creating temporary df for api with only the columns that have in imdb df

df_api = love_movies[['title', 'year', 'imdb_rating', 'director']]
df_api.head()

title  year  \
0  Crazy, Stupid, Love.                                                  2011   
1  Dr. Strangelove or: How I Learned to Stop Worrying and Love the Bomb  1964   
2  Love, Actually                                                        2003   
3  Shakespeare in Love                                                   1998   
4  P.S. I Love You                                                       2007   

   imdb_rating                   director  
0  7.4          Glenn Ficarra, John Requa  
1  8.4          Stanley Kubrick            
2  7.6          Richard Curtis             
3  7.1          John Madden                
4  7.0          Richard LaGravenese

In [25]:
#creating temporary df for imdb to match the columns in df from api
#only excluding column with video links because don't have this info in api

df_imdb = love_movies_imdb[['title', 'year', 'imdb_rating', 'director']]
df_imdb.head()

title  year  \
0  Crazy, Stupid, Love.                                                  2011   
1  Dr. Strangelove or: How I Learned to Stop Worrying and Love the Bomb  1964   
2  Love Actually                                                         2003   
3  Shakespeare in Love                                                   1998   
4  P.S. I Love You                                                       2007   

   imdb_rating                   director  
0  7.4          Glenn Ficarra, John Requa  
1  8.4          Stanley Kubrick            
2  7.6          Richard Curtis             
3  7.1          John Madden                
4  7.0          Richard LaGravenese

- Before comparing making sure that all the columns in both dataframes have the same type


In [26]:
df_api.info()
df_imdb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40 entries, 0 to 39
Data columns (total 4 columns):
title          40 non-null object
year           40 non-null int32
imdb_rating    40 non-null float64
director       40 non-null object
dtypes: float64(1), int32(1), object(2)
memory usage: 1.2+ KB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40 entries, 0 to 39
Data columns (total 4 columns):
title          40 non-null object
year           40 non-null int32
imdb_rating    40 non-null float64
director       40 non-null object
dtypes: float64(1), int32(1), object(2)
memory usage: 1.2+ KB


- checking in general if have any data not matching

In [27]:
df_api == df_imdb

title  year  imdb_rating  director
0   True   True  True         True    
1   True   True  True         True    
2   False  True  True         True    
3   True   True  True         True    
4   True   True  True         True    
5   True   True  True         True    
6   True   True  True         True    
7   True   True  True         True    
8   False  True  True         False   
9   True   True  True         True    
10  True   True  True         True    
11  True   True  True         True    
12  True   True  True         True    
13  True   True  True         True    
14  True   True  True         True    
15  True   True  True         True    
16  True   True  True         True    
17  True   True  True         True    
18  False  True  True         True    
19  True   True  True         True    
20  True   True  True         True    
21  False  True  True         True    
22  True   True  True         False   
23  True   True  True         True    
24  True   True  True         True    
25  True   True  True         True    
26  True   True  True         True    
27  True   True  True         True    
28  True   True  True         True    
29  True   True  True         True    
30  True   True  True         True    
31  True   True  True         True    
32  True   True  True         True    
33  True   True  True         True    
34  True   True  True         True    
35  True   True  True         True    
36  True   True  True         True    
37  True   True  True         True    
38  True   True  True         True    
39  True   True  True         True

- Checking column by column

In [28]:
#checking where titles don't match

df_api.loc[df_api['title'] != df_imdb['title']]
df_imdb.loc[df_api['title'] != df_imdb['title']]

#titles from IMDB are in the original language, not a big problem because have imdb ID to match both dataframes if needed
#can use one title or another depending on the need

title  year  imdb_rating         director
2   Love, Actually        2003  7.6          Richard Curtis 
8   In the Mood for Love  2000  8.1          Kar-Wai Wong   
18  Love Me If You Dare   2003  7.6          Yann Samuell   
21  Show Me Love          1998  7.5          Lukas Moodysson

title  year  imdb_rating                        director
2   Love Actually     2003  7.6          Richard Curtis                
8   Faa yeung nin wa  2000  8.1          Kar-Wai Wong (as Kar Wai Wong)
18  Jeux d'enfants    2003  7.6          Yann Samuell                  
21  Fucking Åmål      1998  7.5          Lukas Moodysson

In [29]:
#checking where years don't match

df_api.loc[df_api['year'] != df_imdb['year']]

#years match in all cases

Empty DataFrame
Columns: [title, year, imdb_rating, director]
Index: []

In [30]:
#checking where imdb_ratings don't match

df_api.loc[df_api['imdb_rating'] != df_imdb['imdb_rating']]

#imdb_ratings match in all cases

Empty DataFrame
Columns: [title, year, imdb_rating, director]
Index: []

In [31]:
#checking where directors don't match

df_api.loc[df_api['director'] != df_imdb['director']]
df_imdb.loc[df_api['director'] != df_imdb['director']]

#imdb only shows two main directors in the page that we are scraping 
#could consider for the future to modify the code to extract the rest of the info (from the hyperlink) 
#but at the moment have this information from API

title  year  imdb_rating  \
8   In the Mood for Love  2000  8.1           
22  New York, I Love You  2008  6.3           

                                                                                                                                                    director  
8   Kar-Wai Wong                                                                                                                                              
22  Fatih Akin, Yvan Attal, Randall Balsmeyer, Allen Hughes, Shunji Iwai, Wen Jiang, Shekhar Kapur, Joshua Marston, Mira Nair, Natalie Portman, Brett Ratner

title  year  imdb_rating                        director
8   Faa yeung nin wa      2000  8.1          Kar-Wai Wong (as Kar Wai Wong)
22  New York, I Love You  2008  6.3          Fatih Akin, Yvan Attal |

After checking 4 main columns can see that the data extracted from the API is precise enough for our needs

### Creating one dataframe with the most important info from API and IMDB, saving to CSV

In [32]:
#take the majority of the columns from the API dataframe
#adding one column with video links from the imdb dataframe 

main_info = love_movies[['title', 'director', 'imdb_rating', 'released_spain', 'actors', 'imdb_id', 'poster']]
main_info['video_link'] = love_movies_imdb['video_link']
main_info.head()

C:\Users\krist\Anaconda3\lib\site-packages\ipykernel_launcher.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """


title  \
0  Crazy, Stupid, Love.                                                   
1  Dr. Strangelove or: How I Learned to Stop Worrying and Love the Bomb   
2  Love, Actually                                                         
3  Shakespeare in Love                                                    
4  P.S. I Love You                                                        

                    director  imdb_rating released_spain  \
0  Glenn Ficarra, John Requa  7.4         2011-07-29       
1  Stanley Kubrick            8.4         1964-01-29       
2  Richard Curtis             7.6         2003-11-14       
3  John Madden                7.1         1999-01-08       
4  Richard LaGravenese        7.0         2007-12-21       

                                                         actors    imdb_id  \
0  Steve Carell, Ryan Gosling, Julianne Moore, Emma Stone        tt1570728   
1  Peter Sellers, George C. Scott, Sterling Hayden, Keenan Wynn  tt0057012   
2  Bill Nighy, Gregor Fisher, Rory MacGregor, Colin Firth        tt0314331   
3  Geoffrey Rush, Tom Wilkinson, Steven O'Donnell, Tim McMullan  tt0138097   
4  Hilary Swank, Gerard Butler, Lisa Kudrow, Gina Gershon        tt0431308   

                                                                                                                               poster  \
0  https://m.media-amazon.com/images/M/MV5BMTg2MjkwMTM0NF5BMl5BanBnXkFtZTcwMzc4NDg2NQ@@._V1_SX300.jpg                                   
1  https://m.media-amazon.com/images/M/MV5BZWI3ZTMxNjctMjdlNS00NmUwLWFiM2YtZDUyY2I3N2MxYTE0XkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SX300.jpg   
2  https://m.media-amazon.com/images/M/MV5BMTY4NjQ5NDc0Nl5BMl5BanBnXkFtZTYwNjk5NDM3._V1_SX300.jpg                                       
3  https://m.media-amazon.com/images/M/MV5BM2ZkNjM5MjEtNTBlMC00OTI5LTgyYmEtZDljMzNmNzhiNzY0XkEyXkFqcGdeQXVyNDYyMDk5MTU@._V1_SX300.jpg   
4  https://m.media-amazon.com/images/M/MV5BNTg2MDg4MjI5NV5BMl5BanBnXkFtZTcwMzQ0MDczMw@@._V1_SX300.jpg                                   

                                                           video_link  
0  https://www.imdb.com/title/tt1570728/videogallery/?ref_=tt_ql_pv_2  
1  https://www.imdb.com/title/tt1570728/videogallery/?ref_=tt_ql_pv_2  
2  https://www.imdb.com/title/tt1570728/videogallery/?ref_=tt_ql_pv_2  
3  https://www.imdb.com/title/tt1570728/videogallery/?ref_=tt_ql_pv_2  
4  https://www.imdb.com/title/tt1570728/videogallery/?ref_=tt_ql_pv_2

In [33]:
main_info.to_csv("/Users/krist/Ironhack_Labs/DIC19-Kristina-project-web/your_code/final_info.csv", index=False)

## CORRELATION BETWEEN RATING AND VALENTINE'S DAY <br>

- Working with the API dataframe
- Calculate days until Valentines day
- See if ratings are higher when the release date is closer to the valentines day (correlation)

In [34]:
#checking the dataframe from API

love_movies.head()

title  year  \
0  Crazy, Stupid, Love.                                                  2011   
1  Dr. Strangelove or: How I Learned to Stop Worrying and Love the Bomb  1964   
2  Love, Actually                                                        2003   
3  Shakespeare in Love                                                   1998   
4  P.S. I Love You                                                       2007   

   rated released_spain  runtime                            genre  \
0  PG-13 2011-07-29      118 min  Comedy, Drama, Romance            
1  PG    1964-01-29      95 min   Comedy                            
2  R     2003-11-14      135 min  Comedy, Drama, Romance            
3  R     1999-01-08      123 min  Comedy, Drama, History, Romance   
4  PG-13 2007-12-21      126 min  Comedy, Drama, Romance            

                    director  \
0  Glenn Ficarra, John Requa   
1  Stanley Kubrick             
2  Richard Curtis              
3  John Madden                 
4  Richard LaGravenese         

                                                                                                                                   writer  \
0  Dan Fogelman                                                                                                                             
1  Stanley Kubrick (screenplay), Terry Southern (screenplay), Peter George (screenplay), Peter George (based on the book: "Red Alert" by)   
2  Richard Curtis                                                                                                                           
3  Marc Norman, Tom Stoppard                                                                                                                
4  Richard LaGravenese (screenplay), Steven Rogers (screenplay), Cecelia Ahern (novel)                                                      

                                                         actors  \
0  Steve Carell, Ryan Gosling, Julianne Moore, Emma Stone         
1  Peter Sellers, George C. Scott, Sterling Hayden, Keenan Wynn   
2  Bill Nighy, Gregor Fisher, Rory MacGregor, Colin Firth         
3  Geoffrey Rush, Tom Wilkinson, Steven O'Donnell, Tim McMullan   
4  Hilary Swank, Gerard Butler, Lisa Kudrow, Gina Gershon         

                                                                                                                                                                                                       plot  \
0  A middle-aged husband's life changes dramatically when his wife asks him for a divorce. He seeks to rediscover his manhood with the help of a newfound friend, Jacob, learning to pick up girls at bars.   
1  An insane general triggers a path to nuclear holocaust that a War Room full of politicians and generals frantically tries to stop.                                                                         
2  Follows the lives of eight very different couples in dealing with their love lives in various loosely interrelated tales all set during a frantic month before Christmas in London, England.               
3  The world's greatest ever playwright, William Shakespeare, is young, out of ideas and short of cash, but meets his ideal woman and is inspired to write one of his most famous plays.                      
4  A young widow discovers that her late husband has left her 10 messages intended to help ease her pain and start a new life.                                                                                

                      language          country  \
0  English                      USA               
1  English, Russian             UK, USA           
2  English, Portuguese, French  UK, USA, France   
3  English                      USA, UK           
4  English                      USA               

                                                             awards  \
0  Nominated for 1 Golden Globe. Another 3 wins & 22 nominations.     
1  Nominated for 4 Oscars. An

### Getting the number of days to Valentine's day per movie

In [35]:
#getting the day of the year when the movie was released

love_movies['year_day_released'] = love_movies['released_spain'].dt.dayofyear

In [36]:
#calculating the number of days from release date to valentine's day
#Valentine's day is day 45 of the year

love_movies['days_to_valentine'] = love_movies['year_day_released'].apply(lambda x: (365-x+45) if x > 45 
                                                                               else ((45-x) if x < 45 else 0))

In [37]:
#checking data types per column

love_movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40 entries, 0 to 39
Data columns (total 27 columns):
title                 40 non-null object
year                  40 non-null int32
rated                 40 non-null object
released_spain        40 non-null datetime64[ns]
runtime               40 non-null object
genre                 40 non-null object
director              40 non-null object
writer                40 non-null object
actors                40 non-null object
plot                  40 non-null object
language              40 non-null object
country               40 non-null object
awards                37 non-null object
poster                40 non-null object
ratings               40 non-null object
metascore             38 non-null object
imdb_rating           40 non-null float64
imdb_votes            40 non-null object
imdb_id               40 non-null object
type                  40 non-null object
dvd                   40 non-null object
boxoffice             22 non

In [38]:
#converting imdb_rating column to float type

love_movies['imdb_rating'] = love_movies['imdb_rating'].astype('float')

### Binning the data based on the days to Valentine's day

In [39]:
day_labels = ['very_close', 'close', 'medium_close', 'far_away', 'very_far_away']
love_movies['day_to_valentine_bins'] = pd.cut(love_movies['days_to_valentine'], 5, labels=day_labels) #5 bins, has to match with how many labels have in the list
love_movies.head()

title  year  \
0  Crazy, Stupid, Love.                                                  2011   
1  Dr. Strangelove or: How I Learned to Stop Worrying and Love the Bomb  1964   
2  Love, Actually                                                        2003   
3  Shakespeare in Love                                                   1998   
4  P.S. I Love You                                                       2007   

   rated released_spain  runtime                            genre  \
0  PG-13 2011-07-29      118 min  Comedy, Drama, Romance            
1  PG    1964-01-29      95 min   Comedy                            
2  R     2003-11-14      135 min  Comedy, Drama, Romance            
3  R     1999-01-08      123 min  Comedy, Drama, History, Romance   
4  PG-13 2007-12-21      126 min  Comedy, Drama, Romance            

                    director  \
0  Glenn Ficarra, John Requa   
1  Stanley Kubrick             
2  Richard Curtis              
3  John Madden                 
4  Richard LaGravenese         

                                                                                                                                   writer  \
0  Dan Fogelman                                                                                                                             
1  Stanley Kubrick (screenplay), Terry Southern (screenplay), Peter George (screenplay), Peter George (based on the book: "Red Alert" by)   
2  Richard Curtis                                                                                                                           
3  Marc Norman, Tom Stoppard                                                                                                                
4  Richard LaGravenese (screenplay), Steven Rogers (screenplay), Cecelia Ahern (novel)                                                      

                                                         actors  \
0  Steve Carell, Ryan Gosling, Julianne Moore, Emma Stone         
1  Peter Sellers, George C. Scott, Sterling Hayden, Keenan Wynn   
2  Bill Nighy, Gregor Fisher, Rory MacGregor, Colin Firth         
3  Geoffrey Rush, Tom Wilkinson, Steven O'Donnell, Tim McMullan   
4  Hilary Swank, Gerard Butler, Lisa Kudrow, Gina Gershon         

                                                                                                                                                                                                       plot  \
0  A middle-aged husband's life changes dramatically when his wife asks him for a divorce. He seeks to rediscover his manhood with the help of a newfound friend, Jacob, learning to pick up girls at bars.   
1  An insane general triggers a path to nuclear holocaust that a War Room full of politicians and generals frantically tries to stop.                                                                         
2  Follows the lives of eight very different couples in dealing with their love lives in various loosely interrelated tales all set during a frantic month before Christmas in London, England.               
3  The world's greatest ever playwright, William Shakespeare, is young, out of ideas and short of cash, but meets his ideal woman and is inspired to write one of his most famous plays.                      
4  A young widow discovers that her late husband has left her 10 messages intended to help ease her pain and start a new life.                                                                                

                      language          country  \
0  English                      USA               
1  English, Russian             UK, USA           
2  English, Portuguese, French  UK, USA, France   
3  English                      USA, UK           
4  English                      USA               

                                                             awards  \
0  Nominated for 1 Golden Globe. Another 3 wins & 22 nominations.     
1  Nominated for 4 Oscars. An

### Checking the correlation between the rating and the days to Valentine's day

- Grouping Days to Valentine's day by the average rating

In [40]:
#grouping

colnames = ['ave_imdb_rating']
df = pd.DataFrame(love_movies.groupby('day_to_valentine_bins')['imdb_rating'].mean().round(2))
df.columns = colnames
df

ave_imdb_rating
day_to_valentine_bins                 
very_close             6.70           
close                  6.89           
medium_close           6.52           
far_away               6.08           
very_far_away          7.12

- Checking the correlation

In [41]:
love_movies['days_to_valentine'].corr(love_movies['imdb_rating']).round(3)

0.048

In [42]:
"""
Low degree: When the value lies below +- 0.29, then it is said to be a small correlation.

As one variable increases, the other variable tends to also increase (if positive) or decrease (if negative).
(if days to valentines increases, then rating decreases, almost no correlation)

"""

'\nLow degree: When the value lies below +- 0.29, then it is said to be a small correlation.\n\nAs one variable increases, the other variable tends to also increase (if positive) or decrease (if negative).\n(if days to valentines increases, then rating decreases, almost no correlation)\n\n'

### Checking the % of the movies that have been release very close - close to Valentine's day 

In [43]:
#checking how many movies have in each bin

love_movies.groupby('day_to_valentine_bins').imdb_rating.count()

day_to_valentine_bins
very_close       9 
close            10
medium_close     10
far_away         5 
very_far_away    6 
Name: imdb_rating, dtype: int64

In [44]:
close_to_valentines = love_movies.loc[(love_movies.day_to_valentine_bins.isin(['very_close', 'close']))].day_to_valentine_bins.count()

print(f"{close_to_valentines/len(love_movies)*100}% of love movies (from this dataset) have been released very close or close to Valentine's day")

47.5% of love movies (from this dataset) have been released very close or close to Valentine's day


## Results

- Collected all the information that was necessary for the campaign, extracted video links from IMDB and downloaded movie posters to a separate folder
- It's possible to select (if needed) which title want, original or English
- It's possible to decide how many movies want to check (by changing the number of pages in the API request)
- Understood the difference in the release year that was given by API
- From checking average rating per bin understood that the average rating is not higher if the release date is closer to the Valentine's day
- Correlation between the ratings and the Valentine's day is very low. As a next step, could check only the movies release in the last few years, maybe there is a different trend latelly. Also, to really see the correlation, would need to ahve a bigger database (have information of more than 40 movies)
- Understood that 47.5% of love movies (from this dataset) have been released very close or close to Valentine's day